src/runtime/gravity_core.c

```diff
@@ -2058,6 +2058,70 @@ static bool string_split (gravity_vm *vm, gravity_value_t *args, uint16_t nargs,
 	RETURN_VALUE(VALUE_FROM_OBJECT(list), rindex);
 }
 
+static bool string_loop (gravity_vm *vm, gravity_value_t *args, uint16_t nargs, uint32_t rindex) {
+	if (nargs < 2) RETURN_ERROR("Incorrect number of arguments.");
+	if (!VALUE_ISA_CLOSURE(GET_VALUE(1))) RETURN_ERROR("Argument must be a Closure.");
+
+	gravity_closure_t *closure = VALUE_AS_CLOSURE(GET_VALUE(1));	// closure to execute
+	gravity_value_t value = GET_VALUE(0);							// self parameter
+	gravity_string_t *string = VALUE_AS_STRING(value);
+	char *str = string->s;
+	register gravity_int_t n = string->len;  // Times to execute the loop
+	register gravity_int_t i = 0;
```

Comment:
I'm a little bit confused about this. You have this variable named `i` and in the for loop you create an scoped variable name `i`. Did that even compile? Consider renaming one of them, for code readability

Reply:
Wow, you are absolutely right. It did actually compile.

Since this can be a learning moment for all since this seems like something that shouldn't compile, I looked at it and apparently (at least in some cases) it is valid, but you will only have access to the `i` variable that was declared in the for loop's scope.

Consider this c example:

```c
#include <stdio.h>

int main() {
  int i = 5;
  for (int i = 0; i < 10; i++) {
    printf("%d", i);
  }
  printf("\n%d\n", i);
  return 0;
}
```

The output is:

```
0123456789
5
```

As the loop was only using the `i` that was in the for loop declaration, but when I exited the for loop that for loop `i` variable was no longer in scope, so it used the other `i`.

Very strange. But while it does work I agree with you it is confusing, so I will change it.